src/Migration.ts

```diff
@@ -0,0 +1,523 @@
+	 *
+	 * @returns A migration document instance.
+	 */
+	createDocumentFromPrismic<TType extends TDocuments["type"]>(
+		document: ExtractDocumentType<ExistingPrismicDocument<TDocuments>, TType>,
+		title: string,
+	): PrismicMigrationDocument<ExtractDocumentType<TDocuments, TType>> {
+		const doc = new PrismicMigrationDocument(
+			this.#migratePrismicDocumentData({
+				type: document.type,
+				lang: document.lang,
+				uid: document.uid,
+				tags: document.tags,
+				data: document.data,
+			}) as PendingPrismicDocument<ExtractDocumentType<TDocuments, TType>>,
+			title,
+			{ originalPrismicDocument: document },
+		)
+
+		this._documents.push(doc)
+
+		return doc
+	}
+
+	/**
+	 * Queries a document from the migration instance with a specific UID and
+	 * custom type.
+	 *
+	 * @example
+	 *
+	 * ```ts
+	 * const contentRelationship = migration.createContentRelationship(() =>
+	 * 	migration.getByUID("blog_post", "my-first-post"),
+	 * )
+	 * ```
+	 *
+	 * @typeParam TType - Type of the Prismic document returned.
+	 *
+	 * @param type - The API ID of the document's custom type.
+	 * @param uid - The UID of the document.
+	 *
+	 * @returns The migration document instance with a UID matching the `uid`
+	 *   parameter, if a matching document is found.
+	 */
+	getByUID<TType extends TDocuments["type"]>(
+		type: TType,
+		uid: string,
+	):
+		| PrismicMigrationDocument<ExtractDocumentType<TDocuments, TType>>
+		| undefined {
+		return this._documents.find(
+			(
+				doc,
+			): doc is PrismicMigrationDocument<
+				ExtractDocumentType<TDocuments, TType>
+			> => doc.document.type === type && doc.document.uid === uid,
+		)
+	}
+
+	/**
+	 * Queries a singleton document from the migration instance for a specific
+	 * custom type.
+	 *
+	 * @example
+	 *
+	 * ```ts
+	 * const contentRelationship = migration.createContentRelationship(() =>
+	 * 	migration.getSingle("settings"),
+	 * )
+	 * ```
+	 *
+	 * @typeParam TType - Type of the Prismic document returned.
+	 *
+	 * @param type - The API ID of the singleton custom type.
+	 *
+	 * @returns The migration document instance for the custom type, if a matching
+	 *   document is found.
+	 */
+	getSingle<TType extends TDocuments["type"]>(
+		type: TType,
+	):
+		| PrismicMigrationDocument<ExtractDocumentType<TDocuments, TType>>
+		| undefined {
+		return this._documents.find(
+			(
+				doc,
+			): doc is PrismicMigrationDocument<
+				ExtractDocumentType<TDocuments, TType>
+			> => doc.document.type === type,
+		)
+	}
+
+	/**
+	 * Migrates a Prismic document data from another repository so that it can be
+	 * created through the current repository's Migration API.
+	 *
+	 * @param input - The Prismic document data to migrate.
+	 *
+	 * @returns The migrated Prismic document data.
+	 */
+	#migratePrismicDocumentData(input: unknown): unknown {
+		if (is.filledContentRelationship(input)) {
+			if (input.isBroken) {
+				return {
+					link_type: LinkType.Document,
+					id: "__broken__",
```

Comment:
❓ #ask: I had an issue with the production Migration API where it needed a 16 character `id` value. Does `"__broken__"` work? I used `"_____broken_____"`.

The error ultimately came from here: https://github.com/prismicio/prismic-types-internal/blob/1d32ee9a36c65af0dca339568ccdee13dfb5e318/src/utils/DocumentId.ts#L7

Reply:
OK I was confused why so many underscores, I'll revert and add a comment.